src/js/stores/middlewares/messages.js

```diff
@@ -0,0 +1,24 @@
+import { ADD_MESSAGE } from 'actions/conversation-actions';
+import { showSettingsNotification } from 'actions/app-state-actions';
+
+
+export function firstMessage(store) {
+    return function(next) {
+        return function(action) {
+            let result = next(action);
+            let state = store.getState();
+            if (state.appState.settingsEnabled && action.type === ADD_MESSAGE && action.message.role === 'appUser' && !state.user.email) {
```

Comment:
I was once the fence about this one really... either a middleware or in the service directly. I think the service is better now.. I think it's better to run this code everytime the user sends a message than everytime an action is dispatched.